Symantec_docs_code.md

```diff
@@ -0,0 +1,112 @@
+
+## Version control and workflow
+
+The overall workflow looked like this, after environment setup:
+
+* Get latest code files from version control (Perforce)
+* Edit or create Javadoc comments, package-info files
+* Edit main page content as necessary (for example, when we added new services)
+* Check out changed files as needed
+* Submit changes for code review
+* Respond to any code review comments/edit
+* Build with writer changes on TeamCity
+* Check Javadoc build out of version control (this could happen earlier or later, too, but obviously before checking new build back in)
+* After TeamCity build passed, manually build Javadoc
+* Check Javadoc build for obvious errors, fix, rebuild
+* Check code files and doc build back into version control, merging diffs as needed
+
+Version control was "optimistic," with multiple people working on the same files at the same time (see preceding list). This could occasionally produce gnarly merges, especially because writers are used to working on content over a period of time. But generally it wasn't a big problem. I would keep the Javadoc main page checked out for a while because I was the only person working on it, but generally I'd try to work on (and therefore check out) a small set of code files that I could manage through to checkin before anyone else needed to work on them.
+
+## Benefits
+
+There were two different kinds of benefit, as I see it:
+
+* Having a writer work on code documentation meant that the docs were better focused on the needs of an external audience. Our entire doc team worked on different aspects of the product as primary user advocate -- code docs were no exception.
+
+* Having the web services docs tightly integrated into the code as part of the comments meant that they always got the attention of the entire product team.
+
+The biggest overall win, though, came from the communication between dev and doc that sharing a workflow created. There simply couldn't be such a thing as an undocumented feature. And in some cases writer questions about what was going on with an error message, for example, produced better error handling generally. I was also able to catch some potential security issues, for example when a set of enums was initially called from the implementation code, potentially exposing black box functionality. The enums had to be documented for the web services, and so the team reluctantly realized that we'd have to violate DRY principles to expose the enums appropriately.
+
+## Cautionary tales
+
+Doc management was continually concerned that putting a writer into the code -- writing code comments and code samples -- would set developer expectations of the writers too high. In other words, it was seen as an un-extensible model. But the web services project was pretty well contained (I continued to work on other aspects of the server docs), and in fact there was quite a lot of code savvy on our docs team as a whole that management had previously been unaware of.
```

Comment:
WOW. This is a great bit of insight. "... would set developer expectations of the writers too high." That said, I think the true concern was the "un-extensible model" -- but there could be a couple of interpretations here. For example "we couldn't afford writers technical enough to do this work" as un-extensible, or maybe "we don't want the developers to think they can offload their work" which is slightly different. Which is it?